knack/arguments.py

```diff
@@ -87,63 +124,82 @@ def get_cli_argument(self, command, name):
 
 
 class ArgumentsContext(object):
-    def __init__(self, command_loader, command):
+    def __init__(self, command_loader, command_scope):
+        """ Context manager to register arguments
+
+        :param command_loader: The command loader that arguments should be registered into
+        :type command_loader: knack.commands.CLICommandsLoader
+        :param command_scope: The command scope this context manager should have
+        :type command_scope: str
+        """
         self.command_loader = command_loader
-        self.commmand = command
+        self.command_scope = command_scope
 
     def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         pass
 
-    def ignore(self, argument_name):
-        self.command_loader.argument_registry.register_cli_argument(self.commmand,
-                                                                    argument_name,
-                                                                    ignore_type)
-
-    def argument(self, argument_name, arg_type=None, **kwargs):
-        self.command_loader.argument_registry.register_cli_argument(self.commmand,
-                                                                    argument_name,
+    def argument(self, argument_dest, arg_type=None, **kwargs):
+        """ Register an argument for the given command scope using a knack.arguments.CLIArgumentType
+
+        :param argument_dest: The destination argument to add this argument type to
+        :type argument_dest: str
+        :param arg_type: The argument type we are registering
```

Comment:
This makes it sound like you have to register an arg_type, but you really don't (and often don't in fact). Rec: "Predefined CLIArgumentType definition to register, as modified by any provided kwargs."

Reply:
done.